src/pal/src/misc/miscpalapi.cpp

```diff
@@ -94,67 +91,142 @@ PAL_GetPALDirectoryW( OUT LPWSTR lpDirectoryName, IN UINT cchDirectoryName )
         /* The path that we return is required to have
            the trailing slash on the end.*/
         lpEndPoint++;
-    }
-    if ( lpFullPathAndName && lpEndPoint && *lpEndPoint != '\0' )
-    {
-        while ( cchDirectoryName - 1 && lpFullPathAndName != lpEndPoint )
-        {
-            *lpDirectoryName = *lpFullPathAndName;
-            lpFullPathAndName++;
-            lpDirectoryName++;
-            cchDirectoryName--;
-        }
-            
-        if ( lpFullPathAndName == lpEndPoint )
-        {
-            *lpDirectoryName = '\0';
-            bRet = TRUE;
-            goto EXIT;
-        }
-        else
+        *lpEndPoint = '\0';
```

Comment:
This modifies the module->lib_name as a side effect, removing the file name from it. We cannot do that, since the returned module is not a copy.